pkg/grafana/synthetic-monitoring-handler.go

```diff
@@ -57,96 +53,48 @@ func (h *SyntheticMonitoringHandler) GetExtension() string {
 	return "json"
 }
 
-func (h *SyntheticMonitoringHandler) newCheckResource(path string, filename string, check Check) grizzly.Resource {
-	resource := grizzly.Resource{
-		UID:      check.UID(),
-		Filename: filename,
-		Handler:  h,
-		Detail:   check,
-		JSONPath: path,
-	}
-	return resource
-}
-
 // Parse parses a manifest object into a struct for this resource type
 func (h *SyntheticMonitoringHandler) Parse(m manifest.Manifest) (grizzly.ResourceList, error) {
-	resources := grizzly.ResourceList{}
-	spec := m["spec"].(map[string]interface{})
-	check := Check{}
-	err := mapstructure.Decode(spec, &check)
-	if err != nil {
-		return nil, err
-	}
-	check["job"] = m.Metadata().Name()
-	resource := h.newCheckResource("", "", check)
-	key := resource.Key()
-	resources[key] = resource
-	return resources, nil
+	resource := grizzly.Resource(m)
+	resource.SetSpecString("job", resource.GetMetadata("name"))
+	resource.SetSpecString(folderNameField, resource.GetMetadata("folder"))
+	return resource.AsResourceList(), nil
 }
 
 // Unprepare removes unnecessary elements from a remote resource ready for presentation/comparison
 func (h *SyntheticMonitoringHandler) Unprepare(resource grizzly.Resource) *grizzly.Resource {
-	delete(resource.Detail.(Check), "tenantId")
-	delete(resource.Detail.(Check), "id")
-	delete(resource.Detail.(Check), "modified")
-	delete(resource.Detail.(Check), "created")
+	resource = *(resource.DeleteSpecKey("tenantId"))
+	resource = *(resource.DeleteSpecKey("id"))
+	resource = *(resource.DeleteSpecKey("modified"))
+	resource = *(resource.DeleteSpecKey("created"))
```

Comment:
This looks like we could have a different representation of resource.

```
return resource.asUnprepared()
```

Reply:
interesting, yes. But, it would still need to call out to a func within the handler to actually do it.